### ISSUE TITLE
Settings for analysis with xyz2mol

### ISSUE THREAD

**Issue by NiklasGebauer** [atomicarchitects/symphony#118]
Dear authors,

first of all, congrats for your insightful work and its publication at ICLR 2024. I really enjoyed reading the paper. The thorough analysis of your model and the baselines is very interesting and well executed. Thank you!

I wanted to ask which version and settings of xyz2mol you use for your analysis. You report that 99.99% of the molecules in QM9 are validated. I recently implemented a script that uses the implementation of xyz2mol in RDKit (rdkit.Chem.rdDetermineBonds) for our `schnetpack-gschnet` package and it only finds SMILES for about 95% (~124k) of the molecules.
Did you use the version from RDKit or the script? And which settings/arguments did you use?

Thanks for your help and best regards,
Niklas

**Comment by ameya98** [comment]
Hi Niklas, thank you for the compliments, and sorry for the delayed response. This is a snippet simplified it from our `metrics.py`:
```python
import os
import tqdm
import rdkit.Chem as Chem
from rdkit.Chem import rdDetermineBonds

print(f"rdkit version: {Chem.rdBase.rdkitVersion}")

valid_count = 0
total_count = 0

root_dir = "/Users/ameyad/Documents/spherical-harmonic-net/others/qm9_xyz"
for molecules_file in tqdm.tqdm(os.listdir(root_dir)):
    if not molecules_file.endswith(".xyz"):
        continue

    molecules_file = os.path.join(root_dir, molecules_file)
    
    total_count += 1
    try:
        mol = Chem.MolFromXYZFile(molecules_file)
        mol = Chem.Mol(mol)
        rdDetermineBonds.DetermineBonds(mol, charge=0)
        smi = Chem.MolToSmiles(mol)
    except ValueError:
        continue
    
    valid_count += 1

print(f"Valid molecules: {valid_count}")
print(f"Total molecules: {total_count}")
print(f"Percentage of valid molecules: {valid_count / total_count * 100:.2f}%")
```
that gives me the following output, which we reported in our paper.
```

**Comment by NiklasGebauer** [comment]
Dear @ameya98 ,

thanks a lot for the clarification! It seems these are re-optimized structures as reported [here](https://www.nature.com/articles/s41597-019-0121-7). I think they are using the same level of theory as originally in QM9 for optimization of the geometries but with a newer version of the Gaussian software. It should not make a big difference, though. You can find the original QM9 structures [here](https://springernature.figshare.com/articles/dataset/Data_for_6095_constitutional_isomers_of_C7H10O2/1057646?backTo=/collections/Quantum_chemistry_structures_and_properties_of_134_kilo_molecules/978904).

I think the main difference comes from the version of rdkit.
I downloaded the files you linked and executed your code snippet on my machine and I still get a lower valitidy:
'''
rdkit version: 2023.09.5
100%|██████████| 133631/133631 [00:27<00:00, 4811.96it/s]
Valid molecules: 126800
Total molecules: 133631
Percentage of valid molecules: 94.89%
'''

I checked the rdkit releases and they have changed a few things related to rdDetermineBonds between the versions. When I install your version 2023.03.2, I also get 99.9%. I found that they changed the default method to get the connectivity in version 2023.09 ([see here](https://github.com/rdkit/rdkit/pull/6740)).
It should be possible to obtain the old behavior in version 2023.09.x with `rdDetermineBonds.DetermineBonds(mol, charge=0, useVdw=True). However, I actually get 94.75% when using this option, so it's slightly lower than the default.

I also installed the newest version 2024.03.2 but found the same behavior. I do not get back to the 99.9% except if I set ` `rdDetermineBonds.DetermineBonds(mol, charge=0, useHueckel)`. It prints a million warnings but validates 133608 structures, so almost as much as with your version. Nevertheless, as far as I know, the version you installed (2023.03.2) should not be using Hueckel if we call it with `rdDetermineBonds.DetermineBonds(mol, charge=0)`.
I'm slightly confused about the different results...

TL;DR It seems to be an issue with changes in RDKit. Thanks for providing the code, this helped me to (at least somewhat) pinpoint the source for the different numbers :)

Best,
Niklas

**Comment by ameya98** [comment]
Very interesting, I did not know about these various options! I wonder if/how the results for our paper change depending on the rdkit version. Let me know if you need any other help!
